// ngx-electron/bundles/ngx-electron.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core')) :
    typeof define === 'function' && define.amd ? define('ngx-electron', ['exports', '@angular/core'], factory) :
    (factory((global['ngx-electron'] = {}),global.ng.core));
}(this, (function (exports,core) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var ElectronService = /** @class */ (function () {
        function ElectronService() {
        }
        Object.defineProperty(ElectronService.prototype, "electron", {
            get: /**
             * @return {?}
             */ function () {
                if (!this._electron) {
                    if (window && window.require) {
                        this._electron = window.require('electron');
                        return this._electron;
                    }
                    return null;
                }
                return this._electron;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(ElectronService.prototype, "isElectronApp", {
            /**
             * determines if SPA is running in Electron
             */
            get: /**
             * determines if SPA is running in Electron
             * @return {?}
             */ function () {
                return !!window.navigator.userAgent.match(/Electron/);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(ElectronService.prototype, "isMacOS", {
            get: /**
             * @return {?}
             */ function () {
                return this.isElectronApp && process.platform === 'darwin';
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(ElectronService.prototype, "isWindows", {
            get: /**
             * @return {?}
             */ function () {
                return this.isElectronApp && process.platform === 'win32';
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(ElectronService.prototype, "isLinux", {
            get: /**
             * @return {?}
             */ function () {
                return this.isElectronApp && process.platform === 'linux';
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(ElectronService.prototype, "isX86", {
            get: /**
             * @return {?}
             */ function () {
                return this.isElectronApp && process.arch === 'ia32';
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(ElectronService.prototype, "isX64", {
            get: /**
             * @return {?}
             */ function () {
                return this.isElectronApp && process.arch === 'x64';
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(ElectronService.prototype, "desktopCapturer", {
            get: /**
             * @return {?}
             */ function () {
                return this.electron ? this.electron.desktopCapturer : null;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(ElectronService.prototype, "ipcRenderer", {
            get: /**
             * @return {?}
             */ function () {
                return this.electron ? this.electron.ipcRenderer : null;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(ElectronService.prototype, "remote", {
            get: /**
             * @return {?}
             */ function () {
                return this.electron ? this.electron.remote : null;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(ElectronService.prototype, "webFrame", {
            get: /**
             * @return {?}
             */ function () {
                return this.electron ? this.electron.webFrame : null;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(ElectronService.prototype, "clipboard", {
            get: /**
             * @return {?}
             */ function () {
                return this.electron ? this.electron.clipboard : null;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(ElectronService.prototype, "crashReporter", {
            get: /**
             * @return {?}
             */ function () {
                return this.electron ? this.electron.crashReporter : null;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(ElectronService.prototype, "process", {
            get: /**
             * @return {?}
             */ function () {
                return this.remote ? this.remote.process : null;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(ElectronService.prototype, "nativeImage", {
            get: /**
             * @return {?}
             */ function () {
                return this.electron ? this.electron.nativeImage : null;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(ElectronService.prototype, "screen", {
            get: /**
             * @return {?}
             */ function () {
                return this.electron ? this.electron.screen : null;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(ElectronService.prototype, "shell", {
            get: /**
             * @return {?}
             */ function () {
                return this.electron ? this.electron.shell : null;
            },
            enumerable: true,
            configurable: true
        });
        return ElectronService;
    }());

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation. All rights reserved.
    Licensed under the Apache License, Version 2.0 (the "License"); you may not use
    this file except in compliance with the License. You may obtain a copy of the
    License at http://www.apache.org/licenses/LICENSE-2.0

    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
    KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
    WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
    MERCHANTABLITY OR NON-INFRINGEMENT.

    See the Apache Version 2.0 License for specific language governing permissions
    and limitations under the License.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (b.hasOwnProperty(p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var ElectronServiceRef = /** @class */ (function (_super) {
        __extends(ElectronServiceRef, _super);
        function ElectronServiceRef() {
            return _super.call(this) || this;
        }
        ElectronServiceRef.decorators = [
            { type: core.Injectable }
        ];
        /** @nocollapse */
        ElectronServiceRef.ctorParameters = function () { return []; };
        return ElectronServiceRef;
    }(ElectronService));

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var NgxElectronModule = /** @class */ (function () {
        function NgxElectronModule() {
        }
        NgxElectronModule.decorators = [
            { type: core.NgModule, args: [{
                        declarations: [],
                        exports: [],
                        providers: [{ provide: ElectronService, useClass: ElectronServiceRef }]
                    },] }
        ];
        return NgxElectronModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */

    exports.NgxElectronModule = NgxElectronModule;
    exports.ElectronService = ElectronService;
    exports.ElectronServiceRef = ElectronServiceRef;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
